### ISSUE TITLE
[BUG] VM migration is stuck because destination pod can't attach a migrating volume

### ISSUE THREAD

**Issue by bk201** [longhorn/longhorn#3234]
**Describe the bug**
This was observed during implementing the upgrade feature of Harvester. We want to allow zero-downtime upgrades, thus we perform the upgrade with this flow:

For each node:
- Wait for the previous upgraded node reboot and back.
- Live migrate VMs on the node to other nodes.
- Drain the node.
- Do node upgrade (patch OS, load images, etc).
- Reboot the node.

The issue happens when migrating VMs on the second node, the operation starts just after the first node is back and ready.
![Screen Shot 2021-11-11 at 4 01 14 PM](https://user-images.githubusercontent.com/1691518/141260515-0bdcb965-7125-45af-ba8b-c3add7928e0e.png)

We saw two issues
- When migrating VMs on the second node, KubeVirt attempts to create a new pod on other nodes for existing VMs, but the pod can't attach VM's volume successfully.
- VMs on another node enters `Terminating` status. From instance-manager logs, it seems that instance-manager deletes VM pods.

**To Reproduce**

- Create a Harvester cluster and perform operations as the bug description.

**Expected behavior**
The VM migration should succeed. And non-involved VMs should continue running.

**Log**

[longhorn-support-bundle_baa464a6-ce63-48fa-afdf-1522a14449f8_2021-11-11T07-13-11Z.zip](https://github.com/longhorn/longhorn/files/7518798/longhorn-support-bundle_baa464a6-ce63-48fa-afdf-1522a14449f8_2021-11-11T07-13-11Z.zip)

**Environment:**
 - Longhorn version: `v1.2.2`
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): Helm
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: RKE2
   - Number of management node in the cluster: 3
   - Number of worker node in the cluster: 1
 - Node config
   - OS type and version:
   - CPU per node:
   - Memory per node:
   - Disk type(e.g. SSD/NVMe):
   - Network bandwidth between the nodes:
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): VMs
 - Number of Longhorn volumes in the cluster:

**Additional context**

Env:
```
NAME    STATUS   ROLES                       AGE     VERSION
node1   Ready    control-plane,etcd,master   5h10m   v1.21.5+rke2r1
node2   Ready    control-plane,etcd,master   5h7m    v1.21.5+rke2r1
node3   Ready    <none>                      5h6m    v1.21.5+rke2r1
node4   Ready    control-plane,etcd,master   5h7m    v1.21.5+rke2r1
```

The support bundle is captured after:

- Live-migrate VMs on `node2` (Done)
- Drain `node2` (Done)
- Reboot `node2`
- Live-migrate VMs on node4 (Stuck)

Issues:
- On `node4`, `virt-launcher-test-vm-4-cc4b8` (pvc `test-vm-4-disk-0`, volume: `pvc-411f7c11-aa4c-4737-a4bb-3b180881880f`) is being migrated to `node2` (`node2` is just back and it's resource usage is low)
    - KubeVirt create the pod `virt-launcher-test-vm-4-dlrxj` on `node2` for migration. The pod keeps pending in `ContainerCreating` status.
    ![Screen Shot 2021-11-11 at 3 22 50 PM](https://user-images.githubusercontent.com/1691518/141262708-f4c72c4d-7eff-4148-a626-00efe6ba8a03.png)
- Two VMs on `node3` entering `Terminating` (Node 3 is not touched at all)
    - VM pods: `virt-launcher-test-vm-2-9cbnr` and `virt-launcher-upgrade-repo-hvst-8wqpb`
    - The `longhorn-manager-c77xq` on `node3` has logs:
        ```
        time="2021-11-11T08:19:28Z" level=info msg="Deleted pod virt-launcher-test-vm-2-9cbnr so that Kubernetes will handle remounting volume pvc-666ca661-4494-4912-984a-852e83359a83" controller=longhorn-kubernetes-pod node=node3
        time="2021-11-11T08:19:28Z" level=info msg="Deleted pod virt-launcher-upgrade-repo-hvst-8wqpb so that Kubernetes will handle remounting volume pvc-4750c94a-d5fc-44c6-89e9-38010b345ae9" controller=longhorn-kubernetes-pod node=node3
        time="2021-11-11T08:19:30Z" level=warning msg="Cannot find the instance status in instance manager instance-manager-r-11b44489 for the running instance pvc-411f7c11-aa4c-4737-a4bb-3b180881880f-r-3ecb4d59, will mark the instance as state ERROR"
        time="2021-11-11T08:19:32Z" level=warning msg="Cannot find the instance status in instance manager instance-manager-r-11b44489 for the running instance pvc-411f7c11-aa4c-4737-a4bb-3b180881880f-r-3ecb4d59, will mark the instance as state ERROR"
        ```

**Comment by shuo-wu** [comment]
The fix for issue https://github.com/longhorn/longhorn/issues/3206 probably works for this bug as well.

**Comment by bk201** [comment]
Close this because Longhorn 1.2.3 had been released and shipped in Harvester 1.0.0.
